json.go

```diff
@@ -22,6 +22,20 @@ type Document struct {
 	Status    string `json:"status"`
 }
 
+type Guide struct {
```

Comment:
I guess this is the biggest thing I wanted to discuss – the difference seems very very minor. I could potentially just insert a few `if`'s into the existing code, and may be rename a few things, like `Document` could become something like `DocumentOrGuide` or `AnyDocument`, or is a guide just a `Document{Product: "tutorials", Tag: "current"}`..?